src/samples/permissionhooks/AllowlistPlugin.sol

```diff
@@ -0,0 +1,142 @@
+    mapping(address target => mapping(bytes4 selector => mapping(address account => bool))) public
+        selectorAllowlist;
+
+    error TargetNotAllowed();
+    error SelectorNotAllowed();
+    error NoSelectorSpecified();
+
+    function onInstall(bytes calldata data) external override {
+        AllowlistInit[] memory init = abi.decode(data, (AllowlistInit[]));
+
+        for (uint256 i = 0; i < init.length; i++) {
+            targetAllowlist[init[i].target][msg.sender] = AllowlistEntry(true, init[i].hasSelectorAllowlist);
+
+            if (init[i].hasSelectorAllowlist) {
+                for (uint256 j = 0; j < init[i].selectors.length; j++) {
+                    selectorAllowlist[init[i].target][init[i].selectors[j]][msg.sender] = true;
+                }
+            }
+        }
+    }
+
+    function onUninstall(bytes calldata data) external override {
+        AllowlistInit[] memory init = abi.decode(data, (AllowlistInit[]));
+
+        for (uint256 i = 0; i < init.length; i++) {
+            delete targetAllowlist[init[i].target][msg.sender];
+
+            if (init[i].hasSelectorAllowlist) {
+                for (uint256 j = 0; j < init[i].selectors.length; j++) {
+                    delete selectorAllowlist[init[i].target][init[i].selectors[j]][msg.sender];
+                }
+            }
+        }
+    }
+
+    function setAllowlistTarget(address target, bool allowed, bool hasSelectorAllowlist) external {
+        targetAllowlist[target][msg.sender] = AllowlistEntry(allowed, hasSelectorAllowlist);
+    }
+
+    function setAllowlistSelector(address target, bytes4 selector, bool allowed) external {
+        selectorAllowlist[target][selector][msg.sender] = allowed;
+    }
+
+    function preUserOpValidationHook(uint8 functionId, PackedUserOperation calldata userOp, bytes32)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        if (functionId == uint8(FunctionId.PRE_VALIDATION_HOOK)) {
+            _checkAllowlistCalldata(userOp.callData);
+            return 0;
+        }
+        revert NotImplemented();
+    }
+
+    function preRuntimeValidationHook(uint8 functionId, address, uint256, bytes calldata data, bytes calldata)
+        external
+        view
+        override
+    {
+        if (functionId == uint8(FunctionId.PRE_VALIDATION_HOOK)) {
+            _checkAllowlistCalldata(data);
+            return;
+        }
+
+        revert NotImplemented();
+    }
+
+    function pluginMetadata() external pure override returns (PluginMetadata memory) {
+        PluginMetadata memory metadata;
+        metadata.name = "Allowlist Plugin";
+        metadata.version = "v0.0.1";
+        metadata.author = "ERC-6900 Working Group";
+
+        return metadata;
+    }
+
+    // solhint-disable-next-line no-empty-blocks
+    function pluginManifest() external pure override returns (PluginManifest memory) {}
```

Comment:
Is this blank because it's supposed to only be installed with a user-provided config?

Reply:
Yep, basically. The manifest is empty because this plugin doesn't define anything that the manifest can install:
- execution functions
- execution hooks
- validation functions
- signature validators
- call permissions